packages/transformer/src/IModelExporter.ts

```diff
@@ -220,26 +239,54 @@ export class IModelExporter {
-  private _excludedElementAspectClasses = new Set<typeof ElementAspect>();
-  /** The set of classFullNames for ElementAspects that will be excluded from transformation to the target iModel. */
-  private _excludedElementAspectClassFullNames = new Set<string>();
   /** The set of classes of Relationships that will be excluded (polymorphically) from transformation to the target iModel. */
   private _excludedRelationshipClasses = new Set<typeof Relationship>();
 
+  /** Strategy for how ElementAspects are exported */
+  private _exportElementAspectsStrategy: ExportElementAspectsStrategy<ElementAspectsHandler>;
+
   /** Construct a new IModelExporter
    * @param sourceDb The source IModelDb
    * @see registerHandler
    */
-  public constructor(sourceDb: IModelDb) {
+  public constructor(sourceDb: IModelDb, elementAspectsStrategy: ElementAspectExportStrategy = ElementAspectExportStrategy.WithElement) {
     this.sourceDb = sourceDb;
+    this._exportElementAspectsStrategy = this.createExportElementAspectsStrategy(sourceDb, elementAspectsStrategy);
+  }
+
+  private createExportElementAspectsStrategy(sourceDb: IModelDb, elementAspectsStrategy: ElementAspectExportStrategy): ExportElementAspectsStrategy<ElementAspectsHandler> {
+    switch(elementAspectsStrategy) {
+      case ElementAspectExportStrategy.WithElement:
+        return new ExportElementAspectsWithElementsStrategy(sourceDb, {
+          trackProgress: async () => this.trackProgress(),
+          onExportElementMultiAspects: (multiAspects) => this.handler.onExportElementMultiAspects(multiAspects),
+          onExportElementUniqueAspect: (uniqueAspect, isUpdate) => this.handler.onExportElementUniqueAspect(uniqueAspect, isUpdate),
+          shouldExportElementAspect: (aspect: ElementAspect) => this.handler.shouldExportElementAspect(aspect),
+        });
+      case ElementAspectExportStrategy.Detached:
+        return new DetachedExportElementAspectsStrategy(sourceDb, {
+          trackProgress: async () => this.trackProgress(),
+          onExportElementMultiAspects: (multiAspects) => this.handler.onExportElementMultiAspects(multiAspects),
+          onExportElementUniqueAspect: (uniqueAspect, isUpdate) => this.handler.onExportElementUniqueAspect(uniqueAspect, isUpdate),
+          shouldExportElementAspect: (aspect: ElementAspect) => this.handler.shouldExportElementAspect(aspect),
+        });
+      default:
+        throw new Error("Unknown ElementAspect exporting strategy");
+    }
   }
 
   /** Register the handler that will be called by IModelExporter. */
   public registerHandler(handler: IModelExportHandler): void {
     this._handler = handler;
   }
 
+  /** Sets ExportElementAspectsStrategy that will be used when exporting ElementAspects.
+   * This method allows to change strategy after IModelExporter was already created.
+   */
+  public setExportElementAspectsStrategy(elementAspectsStrategy: ElementAspectExportStrategy): void {
```

Comment:
is this new API necessary? What happens if you switch from "export per element" to "export all" in the middle? You re-export conflicting aspects?

Reply:
This was only required so that IModelTransformer could set `ElementAspectExportStrategy` based on what parameter it received, but is no longer needed because of your other comment.